### ISSUE TITLE
Allow defining human readable room names for ids

### ISSUE THREAD

**Issue by benedikt-bartscher** [humbertogontijo/homeassistant-roborock#205]
Add a map for name <-> ids to homeassistant-roborock configuration and use these mapped names in all services/attributes

**Comment by kmain4** [comment]
@benedikt-bartscher Can you explain how you're setting up automations with room ids? I can't find any docs

**Comment by Lash-L** [comment]
So this is unfortunately looking to be impossible.

I decided to try to get around to adding this functionality today.

I got about 90% of the way there, when I realized the problem.

HA does not allow dynamically declaring options for services. I can dynamic check the data that someone puts it, but I can't dynamically list them as options.

So unless if I can figure out a way to list the possible rooms so that they can be selected in the clean_segment call, I think this may be DOA.

The only option I could think of would be to create a switch for each room dynamically on start up, but that would spam the device page and really not be that helpful.

**Comment by humbertogontijo** [comment]
@Lash-L What about two entities for this? Select and a button entities?

**Comment by Lash-L** [comment]
As far as I am aware a Select entity is single-item only.  So it would allow for cleaning one room at a time, which I guess is still better than nothing? But I was kind of hoping to be able to have a multi-select kind of deal.

**Comment by humbertogontijo** [comment]
Hmm, thats true

**Comment by Lash-L** [comment]
For now I am just adding the room maps as an extra state variable - which at least partially helps the problem

**Comment by Lash-L** [comment]
I've been thinking - I think for now we can just support cleaning one room for now and in the future add multi room cleaning if a multi select ever gets added (I've created an architecture ticket for a multi support)

So you'd have a select and when you select a room, it will clean it

**Comment by humbertogontijo** [comment]
I'm suspicious that local integration can't clean multiple rooms

**Comment by Lash-L** [comment]
Interesting - I haven't tested. Have your testing shown that?

**Comment by humbertogontijo** [comment]
Yes. But it was along side that issue with the prefixes. I can't remember, but I'm pretty sure it did not accept the dict with room id and repeats

**Comment by Klakurka** [comment]
> So this is unfortunately looking to be impossible.
> 
> I decided to try to get around to adding this functionality today.
> 
> I got about 90% of the way there, when I realized the problem.
> 
> HA does not allow dynamically declaring options for services. I can dynamic check the data that someone puts it, but I can't dynamically list them as options.
> 
> So unless if I can figure out a way to list the possible rooms so that they can be selected in the clean_segment call, I think this may be DOA.
> 
> The only option I could think of would be to create a switch for each room dynamically on start up, but that would spam the device page and really not be that helpful.

Is this now possible as of 2023.07.1 or is this different?

**Comment by Lash-L** [comment]
Core developers want me to built the functionality directly into HA by creating a new entity type. I've been slowly working on it

**Comment by ScottG489** [comment]
Are you referring to this? [Proposal: New multiselect entity type](https://github.com/home-assistant/architecture/discussions/927)

**Comment by Lash-L** [comment]
> Are you referring to this? [Proposal: New multiselect entity type](https://github.com/home-assistant/architecture/discussions/927)


Partially that one!


more specifically:

https://github.com/home-assistant/architecture/discussions/913

and
https://github.com/home-assistant/architecture/discussions/950

**Comment by kbrown01** [comment]
While I understand all of this, I would think very simply providing the names -> room number in attributes would be fine.
I see this in the state:

```
room_numbers:
  '1': null
  '16': null
  '17': null
  '18': null
  '19': null
  '20': null
  '21': null
  '22': null
  '23': null
  '24': null
rooms:
  '1':
    x0: 9150
    y0: 23900
    x1: 13000
    y1: 27050
  '16':
    x0: 18500
    y0: 17850
    x1: 22950
    y1: 24650
....

```

If that had:

```
'16': 'DIning Room'
'17': 'Kitchen'
```
I would be thrilled. I would not need anything more.
Since this is information that is rarely changed, I created a JSON file that has the map id to name and then load that as a REST sensor.  I do see the value here for this discussion as I create the input_booleans by hand, but heck ... that is easy.

```
{
    "rooms": [
        {
            "name": "kitchen",
            "boolean": "input_boolean.vac_kitchen",
            "id": "17"
            },
        {
            "name": "dining_room",
            "boolean": "input_boolean.vac_dining_room",
            "id": "16"
            },
        {
            "name": "foyer",
            "boolean": "input_boolean.vac_foyer",
            "id": "18"
            },
        {
            "name": "laundry",
            "boolean": "input_boolean.vac_laundry",
            "id": "20"
            },
        {
            "name": "living_room",
            "boolean": "input_boolean.vac_living_room",
            "id": "19"
            },
        {
            "name": "master_bedroom",
            "boolean": "input_boolean.vac_master_bedroom",
            "id": "21"
            },
        {
            "name": "master_bathroom",
            "boolean": "input_boolean.vac_master_bathroom",
            "id": "22"
            },
        {
            "name": "guest_bathroom",
            "boolean": "input_boolean.vac_guest_bathroom",
            "id": "23"
            }
    ],
    "points": [
        {
            "name": "sink",
            "x": 17989,
            "y": 25491
         }
    ]
}
```
and

```
##
## Roborock Rooms 
##
- platform: rest
  name: roborock_rooms
  resource: http://192.168.1.245:8123/local/Roborock/roborock.json
  value_template: "{{ now() }}"
  json_attributes:
    - rooms
    - points
```

Then do things like this:
```
alias: Run Vacuum
sequence:
  - service: roborock.vacuum_clean_segment
    data_template:
      segments: |
        {% set vacrooms = namespace(roomid=[]) %}
        {% for rooms in state_attr('sensor.roborock_rooms','rooms') -%}
          {% if is_state(rooms.boolean,'on') %}
            {% set room = rooms.id %}
            {% set vacrooms.roomid = vacrooms.roomid + [room] %}
          {% endif %}
        {%- endfor %}
        {{ vacrooms.roomid }}
    target:
      entity_id: vacuum.roborock_s7
mode: single
icon: mdi:robot-vacuum
```
